ResourceServer/Controllers/ApartmentsController.cs

```diff
@@ -57,35 +59,37 @@ public string Get(int id)
             return JsonConvert.SerializeObject(ap, Formatting.Indented);
         }
 
-        // CREATE POST: api/Apartments
+        // CREATE POST: api/Apartments/add
         [HttpPost("add")]
         public async Task<JObject> Post(Apartment ap)
         {
             var userId = User.FindFirst("sub")?.Value;
             _logger.LogInformation("Adding new apartment owned by " + User.Identity.Name);
 
+            var phoneNum = TrueHomeContext.getPhoneNumber(userId);
+            if (phoneNum == null)
+                TrueHomeContext.setPhoneNumber(ap.PhoneNumber, userId);
+
             ap.IDUser = userId;
             var id = await TrueHomeContext.createApartment(ap);
             return JObject.Parse("{\"id\": " + id + ", \"UploadStatus\": " + 1 + "}");
         }
 
-        // UPDATE PUT: api/Apartments/5
-        [HttpPut("{id}")]
-        public async Task<IActionResult> Put(int id, Apartment ap)
+        // UPDATE PUT: api/Apartments
+        [HttpPut]
+        public async Task<IActionResult> Put(Apartment ap)
         {
-            if (id == ap.ID_Ap)
-            {
-                TrueHomeContext.updateApartment(ap);
-            }
-            return NoContent();
+            TrueHomeContext.updateApartment(ap);
+            return Ok();
         }
 
-        // DELETE: api/ApiWithActions/5
-        [HttpDelete("{id}")]
-        public async Task<IActionResult> Delete(int id)
+        // DELETE: api/Apartments
+        [HttpDelete]
+        public async Task<IActionResult> Delete([FromBody] IDJSON id)
```

Comment:
Przydało by sie kiedyś poczytać czy nie da sie automatycznie bindować tych jednoelementowych JSONów bo to takie bez sensu kapke - to samo sie tyczy zwracania JOSNów tak że ręcznie piszemy stringa. Na pewno jest jakiś framework do tego